src/components/Launch/components/LaunchImage/index.js

```diff
@@ -4,36 +4,46 @@ import "./styles.scss";
 export class LaunchImage extends React.Component {
     constructor(props) {
         super(props);
-        this.state = {//need to add derivedstatefromprops(nextProps, prevState)
-            showImages: this.props.imgBool || false
+        this.state = {
+            isShowingImages: this.props.isShowingImages || false
         }
     }
-    showImagesUpdate = () => {
+    isShowingImagesUpdate = () => {
         this.setState({
-            showImages: !this.state.showImages
+            isShowingImages: !this.state.isShowingImages
         });
     }
+
+    static getDerivedStateFromProps(nextProps, prevState){
+        if(nextProps.isShowingImages!==prevState.isShowingImages){
+            return {isShowingImages: nextProps.isShowingImages}
+        }
+        else{
+            return prevState;
+        }
+    }
     render() {
 
         return (
             <>
                 <img src={this.props.links.mission_patch}
                     alt="mission patch"
                     className="launchImg"></img>
-                {this.props.links.flickr_images.length > 0 &&
+                {(this.state.isShowingImages && this.props.isCurrentRoute===false) &&
```

Comment:
When you have a really long confusing statement like this, it's a good idea to pull this out into a constant at the "render" level.

```
const { isCurrentRoute } = this.props;
const { isShowingImages } = this.state;
const shouldShowButton = isShowingImages && isCurrentRoute === false;
```

Consider destructuring the rest of `this.props` and `this.state` below.